onnxruntime/core/providers/cuda/nn/batch_norm.cc

```diff
@@ -46,6 +48,22 @@ Status BatchNorm<T>::ComputeInternal(OpKernelContext* p_op_kernel_context) const
   auto mean_data = reinterpret_cast<const CudaT*>(mean->template Data<T>());
   auto var_data = reinterpret_cast<const CudaT*>(var->template Data<T>());
 
+  if (X->DataType() == DataTypeImpl::GetType<MLFloat16>()) {
```

Comment:
Run all BN types through customized kernel? Seems no need to go through cudnn below

Reply:
cudnn one is faster for float type, will switch back to cudnn lib once fix the issue using cudnn.